Examples/Python/python/acts/examples/reconstruction.py

```diff
@@ -920,6 +941,39 @@ def addVertexFitting(
 
     customLogLevel = acts.examples.defaultLogging(s, logLevel)
 
+    if trackSelectorRanges is not None:
+        selectedTrackParameters = "trackparameters"
+        trackIndices = "outputTrackIndices"
+
+        trackSelector = acts.examples.TrackSelector(
+            level=customLogLevel(),
+            inputTrackParameters=trackParameters,
+            outputTrackParameters=selectedTrackParameters,
+            outputTrackIndices=trackIndices,
+            **acts.examples.defaultKWArgs(
+                loc0Min=trackSelectorRanges.loc0[0],
+                loc0Max=trackSelectorRanges.loc0[1],
+                loc1Min=trackSelectorRanges.loc1[0],
+                loc1Max=trackSelectorRanges.loc1[1],
+                timeMin=trackSelectorRanges.time[0],
+                timeMax=trackSelectorRanges.time[1],
+                phiMin=trackSelectorRanges.phi[0],
+                phiMax=trackSelectorRanges.phi[1],
+                etaMin=trackSelectorRanges.eta[0],
+                etaMax=trackSelectorRanges.eta[1],
+                absEtaMin=trackSelectorRanges.absEta[0],
+                absEtaMax=trackSelectorRanges.absEta[1],
+                ptMin=trackSelectorRanges.pt[0],
+                ptMax=trackSelectorRanges.pt[1],
+                removeCharged=trackSelectorRanges.removeCharged,
+                removeNeutral=trackSelectorRanges.removeNeutral,
+            ),
+        )
+        s.addAlgorithm(trackSelector)
+    else:
+        selectedTrackParameters = trackParameters
+        trackIndices = ""
```

Comment:
not sure if this makes sense. any suggestions @paulgessinger @timadye

Reply:
If I call with `trackSelectorRanges=None`, so setting `trackIndices=""` I get:
```
You need to either provide collection of truth particles matching 1:1 to tracks, or track indices and all-tips container to do truth matching
```
Maybe the only way to do the truth matching is to use a `TrackSelector` with no cuts, and revert all the changes that allow for `trackSelectorRanges=None`. If so, apologies for suggesting that.

I'm still not clear how this is supposed to work. I assumed that `TrackSelector`'s `outputTrackIndices` gave the mapping between `inputTrackParameters` and `outputTrackParameters`, but `RootVertexPerformanceWriter`'s checks imply that `inputFittedTracksIndices` gives the mapping between `input(All|Selected)TruthParticles` and `inputFittedTracks`.

Reply:
In order to close this PR I would just default to an empty track selection for now. the callee can still overwrite `trackSelectorRanges` if necessary. afterwards I will take a deeper look into `RootVertexPerformanceWriter` and see if and what needs to be adopted

Reply:
@paulgessinger, thanks for the explanation. We certainly don't have a one-to-one match (is that for truth tracking?), so we should keep the indices in all cases. We can just use the no-cuts `TrackSelector` always and make the default `TrackSelectorRanges()`.

ah, I see @andiwand just did that.

Reply:
Just one concern about making the default `TrackSelectorRanges()`: this changes the default behavior to running `TrackSelector`, where before it wasn't run. Will that cause problems for the truth tracking? I don't know if we have any examples of that.

Reply:
ah, I see `addVertexFitting` is also used by `Examples/Scripts/Python/vertex_fitting.py`, which still uses a separate `TrackSelector` call. I guess that should be integrated, and you can put `trackSelectorRanges=None` for the case where it isn't applied.

Reply:
sorry, I'll shut up now and let you finish the other changes and can make a final check when you are done.

Reply:
oh yeah good point about the `vertex_fitting.py` - I will adopt that. hopefully the CI results will recover then